### ISSUE TITLE
Content-Types Builder plugin's order to show collections types not A-Z

### ISSUE THREAD

**Issue by lascapi** [strapi/strapi#8519]
### Describe the bug
The "Content-Types Builder" plugin's order to show collections types. It's not A-Z and not by update or creation

### Steps to reproduce the behavior

1. Go to the "Content-Types Builder"
2. Add a collection
3. The new collection is not in the right place

### Expected behavior

An A-Z order

### Screenshots

![Image](https://img.tedomum.net/data/strapi-a6fa01.png) 

and there [https://strapi.slack.com/archives/C0BNGCDNH/p1604049930115000](https://strapi.slack.com/archives/C0BNGCDNH/p1604049930115000)

### System

- Node.js version: 12.18.0
- NPM version: 6.14.8
- Strapi version: 3.2.4
- Database: Postgres
- Operating system: Windows10

**Comment by derrickmehaffy** [comment]
Confirmed on Strapi v3.2.5

**Comment by akhilmhdh** [comment]
@derrickmehaffy  @soupette  So I have gone through the issue and found the bug. Data that comes to Leftmenu is sorted by using an npm package - [matchSorter](https://www.npmjs.com/package/match-sorter). 

The data is actually sorted according to the key title, which is the Display Name in content types builder
![firstOne](https://user-images.githubusercontent.com/31166322/99772867-92219b80-2b31-11eb-8ddb-f5ba01a50dd2.JPG)

Now the issue was title gets saved as it is, its case sensitive on saving. So with matchSorter it sees address and Address as different and on sorting Address will be the first one. (But address and Address wont be added together, checked already😄 it will fire an error of uniqueness. ) On displaying the names gets formatted to camelCase and that's why we couldn't notice it.

But I have fixed this, from official matchSorter I have made the sorting case insensitive.
